crates/client/src/client/client.rs

```diff
@@ -404,6 +409,18 @@ pub trait Client {
 
         runtime.block_on(client.delete_all(name_of_records, zone_origin, dns_class))
     }
+
+    /// Download all records from a zone using a AXFR query.
+    ///
+    /// # Arguments
+    /// * `zone_origin` - the zone name to update, i.e. SOA name
+    fn zone_transfer(&self, name: &Name) -> ClientResult<Vec<DnsResponse>> {
```

Comment:
I wonder if we want this given the potential for the collect to make an extremely large allocation.

Other option would be to wrap the Stream in an Iterator (I don't know if that's possible) and then allow the downstream consumer to instead work with that. It might not be possible based on what would need to be captured in the Iterator...

Reply:
I don't think the `&mut Context` can be captured to make polling the Stream possible from an Iterator, but it would probably be possible to have a custom iterator calling block_on(stream.next()) in a loop